include/svs/core/distance/cosine.h

```diff
@@ -29,7 +28,12 @@
 #include <iostream>
 #include <memory>
 #include <type_traits>
+
+SVS_VALIDATE_BOOL_ENV(SVS_AVX512_F)
+#if SVS_AVX512_F
```

Comment:
i think it can be generalized to x86 check and not AVX512 specifically?

Reply:
make sense, done